### PR TITLE
feat: utilize biome.js

### PR DESCRIPTION
**Pull request by nexpid**
adds biome.js to CI and adds linting rules

### CODE REVIEW

**Comment by nexpid** [comment]
tested this on my ximi android, nothing's broken

**Comment by PalmDevs** [comment]
Don't we already have [Biome](https://biomejs.dev) for this? Just need to add it to CI and turn off a few rules so nothing explodes.

**Comment by therealolie** [comment]
rejected, uses dos line-endings
(except i cant reject)

**Comment by nexpid** [comment]
> Don't we already have [Biome](https://biomejs.dev) for this? Just need to add it to CI and turn off a few rules so nothing explodes.

oh, had no clue this existed, will switch to it

**Comment by RivenSkaye** [comment]
> rejected, uses dos line-endings
> (except i cant reject)

Most valid reason to stall amd starve a PR

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,8 +28,21 @@ jobs:
...
+        run: bun run format
+
+      - name: Push CI
+        working-directory: main
         run: |
-          bun i
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git add .
+          git commit -m "chore: ci" || exit 0
```
Comment (xHyroM) [inline .github/workflows/build.yml:44]: Isn't `style: format according to standards` or something like that better?

File: package.json [hunk 2]
```diff
@@ -14,7 +16,7 @@
   "author": "maisymoe",
   "license": "BSD-3-Clause",
   "devDependencies": {
-    "@biomejs/biome": "^1.5.3",
+    "@biomejs/biome": "1.5.3",
```
Comment (PalmDevs) [inline package.json:19]: Do we need to limit this to specifically `1.5.3`? `^1.5.3` allows us to still get feature updates, but breaking changes don't count.

**Pull request merged**